AFQ/definitions/utils.py

```diff
@@ -0,0 +1,109 @@
+__all__ = ["Definition", "find_file"]
+
+
+class Definition():
+    '''
+    All Definitions should inherit this.
+    For a given subject and session within the API, the definition is used
+    to create a given mask, map, etc.
+    Definitions have an init function which the users uses to specify
+    how they want the definition to behave.
+    The find_path and get_for_row functions are called by the AFQ API.
+    The api calls find_path to let the definition find relevant files
+    for the given subject and session. The api calls get_for_row to get the
+    mask, map, etc.
+    '''
+
+    def __init__(self):
+        raise NotImplementedError("Use a definition that inherits this class")
```

Comment:
I think this error would be confusing, because it's raised by the inheriting class, not by this class.